smoke-test.sh

```diff
@@ -1,7 +1,72 @@
-#! /usr/bin/env bash
-set -e
+#!/usr/bin/env bash
 
-mkdir /tmp/cargo_raze_scratch && cd /tmp/cargo_raze_scratch
-git clone https://github.com/acmcarther/cargo-raze-examples
-cd cargo-raze-examples
-(cd ./internal && ./evaluate-local-raze.sh)
+set -eu
+
+PWD="$(pwd)"
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd "$DIR"
+
+EXAMPLES_DIR="$DIR/examples"
+TEST_DIR="$DIR/smoke_test"
+
+command -v "cargo-vendor" >/dev/null 2>&1 || ( echo "Cargo Vendor doesn't exist. Please install with `cargo install cargo-vendor`."; exit 1 )
+command -v "bazel" >/dev/null 2>&1 || ( echo "Bazel doesn't exist. Please install using instructions at https://bazel.build/"; exit 1 )
```

Comment:
I added a function, but it ends up looking similar. Because of `set -eu`, non-zero codes must be explicitly caught using `||`, and to prevent multiple `command_exists "X"; if [[ $? -ne 0 ]]; then ...` blocks, I just early exit.

Can change the style if you want it.

Reply:
I don't really feel that strongly about this, so current style sgtm.